backend/mechanismbackend/api/views.py

```diff
@@ -31,9 +31,30 @@ class Meta:
 
 class MechanismList(generics.ListAPIView):
     queryset = Mechanism.objects.all()
-    serializer_class = MechanismSerializer
-    filter_backends = (filters.DjangoFilterBackend,)  # enable filter-backend for this view
-    filter_class = MechanismFilter
+
+    def get_queryset(self, **kwargs):
+        qs = super(MechanismList, self).get_queryset()
+        self.filter = MechanismFilter(self.request.GET, queryset=qs)
+        return self.filter.qs
+
+    def list(self, request):
+        queryset = self.get_queryset()
+        result_list = list(queryset)
+        result_list = self.reorder_by_dissimilarity(result_list)
+        return Response(MechanismSerializer(result_list, many=True).data)
+
+    def reorder_by_dissimilarity(self, mechanisms):
+        for i in range(len(mechanisms)):
+            mech_a = mechanisms[i]
+            max_dissimilarity = 0
+            for k in range(i + 1, len(mechanisms)):
+                mech_b = mechanisms[k]
+                dissimilarity = mech_a.get_dissimilarity(mech_b)
```

Comment:
The dissimilarity should be an average of all dissimilarities to each of the `i` preceding mechanisms. This doesn't mean we need a third loop because mechanism `k` was already compared to `i-1` mechanisms and we just need to add the `i`th comparison. An efficient DP approach could use a quadratic table that stores the sum of all dissimilarities between any two mechanisms. In each iteration, one dissimilarity is added to the corresponding cell. The new value is then divided by `i` (outside the table), which gives us the dissimilarity to all preceding mechanisms.

Reply:
I disagree because for display reasons we just need to maximize the difference to the neighboring mechanisms so we do not need to calculate an average dissimilarity to all other mechanisms

Reply:
I agree that a high dissimilarity to neighboring mechanisms has a higher importance. The neighborhood should still consider more than just two mechanisms. Since maximizing the dissimilarity is the last step of ranking the results, you can consider all mechanisms on the same page as neighbors. I also made a correction to the upper approach, which you can still use to implement this definition of neighborhood. You would just have to clear the table whenever you reach a new page.

Reply:
It actually is not the last step as the ranking is happening before the pagination is applied by the framework. I also do not think we benefit from this approach. I compared the results with and without the current ranking approach and it already yields sufficient results

Reply:
Since we have no runtime problems anymore when we filter for `correct=true`, you implemented an own paginator, and @ThijsRoumen requested to maximize the dissimilarity on each page as well, please implement this by using the approach I described above.